sources/packages/backend/workflow/test/2021-2022/assessment-gateway/assessment-gateway.studentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,55 @@
+import { ASSESSMENT_ID } from "@sims/services/workflow/variables/assessment-gateway";
+import { AssessmentTriggerType } from "@sims/sims-db";
+import { ZBClient } from "zeebe-node";
+import {
+  createFakeConsolidatedFulltimeData,
+  ZeebeMockedClient,
+  E2E_STUDENT_STATUS,
+  PROCESS_INSTANCE_CREATE_TIMEOUT,
+  Workers,
+} from "../../test-utils";
+import { PROGRAM_YEAR } from "../constants/program-year.constants";
+
+describe(`E2E Test Workflow assessment gateway on student appeal for ${PROGRAM_YEAR}`, () => {
+  let zeebeClientProvider: ZBClient;
+  beforeAll(async () => {
+    zeebeClientProvider = ZeebeMockedClient.getMockedZeebeInstance();
+  });
+
+  it("Should follow the expected workflow path when student single and independent.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedFulltimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.assessmentTriggerType =
+      AssessmentTriggerType.StudentAppeal;
+
+    // Act/Assert
+    const assessmentGatewayResponse =
+      await zeebeClientProvider.createProcessInstanceWithResult({
+        bpmnProcessId: "assessment-gateway",
+        variables: {
+          ...assessmentConsolidatedData,
+          [ASSESSMENT_ID]: 1,
+          [E2E_STUDENT_STATUS]: "independentSingleStudent",
+        },
+        requestTimeout: PROCESS_INSTANCE_CREATE_TIMEOUT,
+      });
+    // Workflow instance expected to pass through associate workflow instance worker.
+    expect(
+      assessmentGatewayResponse.variables[Workers.AssociateWorkflowInstance],
```

Comment:
Some workers are reused and then have the same task name, for instance, update-application-status` can be called multiple times during the workflow execution. We can maybe use the service tasks id (also available on worker jobs object) to execute the asserts.